simtools/data_model/validate_data.py

```diff
@@ -221,41 +217,52 @@ def _get_reference_unit(self, column_name):
 
         """
 
-        try:
-            reference_unit = self._reference_data_columns[column_name]["unit"]
-        except KeyError:
-            self._logger.error(
-                f"Data column '{column_name}' not found in reference column definition"
-            )
-            raise
-
+        reference_unit = self._get_reference_data_column(column_name).get("units", None)
         if reference_unit == "dimensionless" or reference_unit is None:
             return u.dimensionless_unscaled
 
         return u.Unit(reference_unit)
 
-    def _column_status(self, col_name):
+    def _check_for_not_a_number(self, col):
```

Comment:
Can't this function be replaced by `np.isfinite`? Maybe not entirely cause you do allow a non-finite number and for some reason return True in that case, but this function can definitely be simplified using `np.isfinite`.

Reply:
Not exactly, as this function allows to
- accept NaN as values (e.g., if there is no value available, we put NaNs) in the data column if this is set in `input_processing::allow_nan`.
- in contrast, I don't see an application to allow infinite numbers.

Reply:
OK!